### PR TITLE
Added Prometheus support.

### PR DESCRIPTION
**Pull request by ewiseblatt**
@lwander or @jtk54 

(essentially my prototype implementation with minor tweaks for the server refactoring)
The json files will come later in the installation PR

### CODE REVIEW

File: spinnaker-monitoring/prometheus_service.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        tags = list(info.tags)
+        all_tags = list(tags)
+        if self.__add_metalabels:
+          all_tags.extend(['job', 'instance'])
+        member = family(member_name, '', labels=all_tags)
+        all_members.append(member)
+
+        for record in info.records:
+          if isinstance(record, dict):
+            print '*** RECORD {0}'.format(record)
```
Comment (lwander) [inline spinnaker-monitoring/prometheus_service.py]: Are these for logging, or a part of the Prometheus interface?

File: spinnaker-monitoring/prometheus_service.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    self.__spectator = spectator_client.SpectatorClient(options)
+    self.__add_metalabels = config.get('prometheus_add_source_metalabels', True)
+    REGISTRY.register(self)
+
+  def __collect_instance_info(
+        self, service, name,
+        instance, metric_metadata, service_metadata, service_to_name_to_info):
+    """Creates a post payload for a DataDog time series data point.
+
+       See http://docs.datadoghq.com/api/?lang=python#metrics-post.
```
Comment (lwander) [inline spinnaker-monitoring/prometheus_service.py]: Why is this in the prometheus support?

File: spinnaker-monitoring/prometheus_service.py (hunk 1 above)
Comment (ewiseblatt) [inline spinnaker-monitoring/prometheus_service.py]: good catch. artifact from development removed.

File: spinnaker-monitoring/prometheus_service.py (hunk 2 above)
Comment (ewiseblatt) [inline spinnaker-monitoring/prometheus_service.py]: good catch. originally copy pasted from datadog implementation to get the outline in place.

**Pull request merged**